source/keyboardHandler.py

```diff
@@ -60,6 +61,14 @@
 #: Whether the sticky NVDA modifier is locked.
 stickyNVDAModifierLocked = False
 
+@contextmanager
+def ignoreInjection():
+	"""Context manager that allows ignoring injected keys temporarily by using a with statement."""
+	global ignoreInjected

```

Comment:
This won't be thread safe. If multiple threads use this context manager at once, the value of ignoreInjected could be squashed.